Use PKG_CHECK_MODULES to detect gpg-error, libassuan and libgcrypt

kloczek opened the pull request:
1. Looks like gnupg-pkcs11-scd is probably last project which still uses *-config scripts to detest some libraries.
Swich to use PKG_CHECK_MODULES().
2. Simplify openssl detestion

Signed-off-by: Tomasz Kłoczko <kloczek@github.com>

alonbl commented on configure.ac:
```diff
@@ -211,49 +190,13 @@ AC_CHECK_FUNCS([ \
-
+PKG_CHECK_MODULES([LIBGPG_ERROR], [gpg-error], [HAVE_LIBGPG_ERROR="yes"])
+PKG_CHECK_MODULES([LIBASSUAN], [libassuan], [HAVE_LIBASSUAN="yes"])
+PKG_CHECK_MODULES([LIBGCRYPT], [libgcrypt], [HAVE_LIBGCRYPT="yes"])
```
these are not optional libraries, we do not need the `HAVE_*` variables, we should just fail.

kloczek replied:
I've modified commit to make those three libraries unconditional.

BTW if gnupg-pkcs11-scd is build with openssl libcrypto it should be possible to use this API instead libgcrypto.
Am I right? 🤔

alonbl reviewed (commented):
Hi,
Thanks!
Have they actual finish the work and is it working properly?
I started[1] the work in the past, but instead of using the standard tools they re-invented the wheel.
Regards,

[1] https://gnupg-devel.gnupg.narkive.com/8IoDVOJf/patch-libgpg-error-gpg-error-m4-support-pkg-config

kloczek committed:
Use PKG_CHECK_MODULES to detect gpg-error, libassuan and libgcrypt

1. Looks like gnupg-pkcs11-scd is probably last project which still uses
   *-config scripts to detest some libraries.
   Swich to use PKG_CHECK_MODULES().
2. Simplify openssl detestion

Signed-off-by: Tomasz Kłoczko <kloczek@github.com>

alonbl commented:
Thanks! Merged.
